libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendMessagingExtensionAuthResponse.ts

```diff
@@ -0,0 +1,187 @@
+            case 'title':
+                return new StringExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the dialog is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional, initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A promise representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: Record<string, unknown>): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled?.getValue(dc.state)) {
+            return dc.endDialog();
+        }
+
+        // Type check that dc.context.adapter matches interface, ExtendedUserTokenProvider
+        // eslint-disable-next-line no-prototype-builtins
+        if (!(typeof (dc.context.adapter as BotFrameworkAdapter)?.getUserToken === 'function')) {
+            throw new Error(`${SendMessagingExtensionAuthResponse.$kind}: not supported by the current adapter.`);
+        }
+
+        const connectionName = this.connectionName?.getValue(dc.state);
+        if (!connectionName) {
+            throw new Error(`${SendMessagingExtensionAuthResponse.$kind} requires a Connection Name.`);
+        }
+
+        const title = this.title?.getValue(dc.state);
+        if (!title) {
+            throw new Error(`${SendMessagingExtensionAuthResponse.$kind} requires a Title.`);
+        }
+
+        const tokenResponse = await SendMessagingExtensionAuthResponse.getUserToken(
+            dc,
+            (dc.context.adapter as unknown) as ExtendedUserTokenProvider,
+            connectionName
+        );
+        if (!tokenResponse) {
+            // We have the auth token, so the user is already signed in.
+            // Similar to OAuthInput, just return the token in the property.
+            if (this.property != null) {
+                dc.state.setValue(this.property?.getValue(dc.state), tokenResponse);
+            }
+
+            // End the dialog and return the token response
+            return dc.endDialog(tokenResponse);
+        }
+
+        // There is no token, so the user has not signed in yet.
+        const activity = await this.getInvokeResponseWithSignInLink(
+            dc,
+            title,
+            (dc.context.adapter as unknown) as ExtendedUserTokenProvider,
+            connectionName
+        );
+        await dc.context.sendActivity(activity);
+
+        // Since a token was not retrieved above, end the turn.
+        return Dialog.EndOfTurn;
+    }
+
+    /**
+     * Builds the compute Id for the dialog.
+     *
+     * @returns {string} A string representing the compute Id.
+     */
+    protected onComputeId(): string {
+        return `${this.constructor.name}[
+            ${this.title?.toString() ?? ''}
```

Comment:
Now that I look a bit closer at these, using the ` character means that newlines are preserved, which I doubt is what you want. In the developer console:

```
> `foo[
bar
]`
> "foo[
bar
]"
```

Reply:
My intent for this one was to ensure an exact match to the class name (especially if it ends up changing before merge). Is there a better way or should I just hard-code it in?